studio/src/components/layout/graph-layout.tsx

```diff
@@ -183,35 +192,75 @@ export const GraphSelect = () => {
 
   const router = useRouter();
   const slug = router.query.slug as string;
-  if (router.pathname.split("/")[2] !== "graph") return null;
+  const namespace = router.query.namespace as string;
+
+  const selected = data?.graphs.find(
+    (g) => g.name === slug && g.namespace === namespace,
+  );
+
+  const groupedGraphs = data?.graphs.reduce(
+    (result, graph) => {
+      const { namespace, name } = graph;
+
+      if (!result[namespace]) {
+        result[namespace] = [];
+      }
+
+      result[namespace].push(graph);
+
+      return result;
+    },
+    {} as Record<string, FederatedGraph[]>,
+  );
+
+  if (router.pathname.split("/")[3] !== "graph") {
```

Comment:
This looks like a workaround. Could you explain and a few comments?

Reply:
I am not sure why this was done. Eelco might know. I am keeping this to not potentially break it